blender_addons/io_md5_exporter/io_export_md5.py

```diff
@@ -993,86 +1013,256 @@ def invoke(self, context, event):
+  textureDirectory = os.path.join(assetDirectory,"textures")
+  if not os.path.exists(textureDirectory):
+    os.makedirs(textureDirectory)
+
+  if os.path.isfile(texturePath2):
+    copy(texturePath2,textureDirectory)
+  else:
+    copy(texturePath, textureDirectory)
+  return{'FINISHED'}
 
 class ExportPrefabForTerasology(bpy.types.Operator):
   bl_idname = "export.prefab"
   bl_label = "Export a prefab for Terasology"
 
   def invoke(self, context, event):
-    exportPrefabForTerasology()
+    assetDirectory = getTargetTerasologyAssetsDirectory()
+    exportPrefabForTerasology(assetDirectory)
+
+    return{'FINISHED'}
+
+class ExportMaterialForTerasology(bpy.types.Operator):
+  bl_idname = "export.material"
+  bl_label = "Export a material for Terasology"
+
+  def invoke(self, context, event):
+    assetDirectory = getTargetTerasologyAssetsDirectory()
+    exportMaterialForTerasology(assetDirectory)
 
     return{'FINISHED'}
 
+class ExportTextureForTerasology(bpy.types.Operator):
+  bl_idname = "export.texture"
+  bl_label = "Export a texture for Terasology"
 
+  def invoke(self, context, event):
+    assetDirectory = getTargetTerasologyAssetsDirectory()
+    exportTextureForTerasology(assetDirectory, self.report)
+
+    return{'FINISHED'}
+
+from tempfile import TemporaryDirectory
+from shutil import copy
+import shutil
+import tempfile
 import zipfile,os
 class ExportZipFileForTerasology(bpy.types.Operator):
+  '''Create a zip that can be directly used in Terasology'''
   bl_idname = "create.zipfile"
   bl_label =  "Create a ZipFile for exporting the content"
-  def invoke(self, context, event):
-    global scale # TODO get rid of it, using a global variable is no good style..
-    scale = 1.0
-    
-    modelFilePath = getTargetTerasologyMeshFilePath(context)
-    animationFilePath = getTargetTerasologyAnimFilePath(context)
-        
-    save_md5(modelFilePath, animationFilePath)
-    exportPrefabForTerasology()
-    assetDirectory = getTargetTerasologyAssetsDirectory(context)
-    dire =  os.path.join(os.path.dirname(assetDirectory), 'assets.zip')
-    zipf = zipfile.ZipFile(dire, 'w', zipfile.ZIP_DEFLATED)
-
-    for root, dirs, files in os.walk(assetDirectory):
-
-        if root.replace(assetDirectory,'') == '':
-          prefix = ''
-        else:
-          prefix = root.replace(assetDirectory, '') + '/'
-          if prefix[0] == '/' :
-            prefix = prefix[1:]
-        for file in files:
-            filePath = root + '/' + file
-            inZipPath = prefix + file
-            zipf.write(filePath, inZipPath)
-    print("Export .zip at ", os.path.dirname(assetDirectory))
-    zipf.close()
 
-    return{'FINISHED'}
+  filepath = StringProperty(subtype = 'FILE_PATH',name="File Path", description="Filepath for exporting", maxlen= 1024, default= "")
+  zipname = StringProperty(name="zip Name", description="Name for the zip to be exported",maxlen=64,default="")

```

Comment:
hmm from just reading the source it seems redundant to have a file path and a zip name.